api/backend/blueflood/blueflood.go

```diff
@@ -132,20 +140,61 @@ func (b *blueflood) FetchSingleSeries(request api.FetchSeriesRequest) (api.Times
 	}
 	queryResolution := b.config.bluefloodResolution(
 		request.Timerange.Resolution(),
-		request.Timerange.Start())
+		request.Timerange.Start(),
+	) // The query resolution will be downsampled automatically based on the timerange?
 
+	// <Sample the data>
 	queryUrl, err := b.constructURL(request, sampler, queryResolution)
 	if err != nil {
 		return api.Timeseries{}, err
 	}
-
-	// Issue GET to fetch metrics
 	parsedResult, err := b.fetch(request, queryUrl)
 	if err != nil {
 		return api.Timeseries{}, err
 	}
+	// </Sample the data>
+
+	// <Sample the full resolution data>
+	fullResolutionRequest := request
+
+	// The "now" as an epoch in milliseconds
+	now := getNow().Unix() * 1000
```

Comment:
Why do we need the current timestamp? Shouldn't we be able to infer the beginning and end range of full resolution data we need to fetch?